src/components/TodoList.js

```diff
@@ -2,19 +2,19 @@ import React from "react";
 import PropTypes from "prop-types";
 import TodoListItem from "./TodoListItem";
 
-const TodoList = ({ todoList, onRemoveTodo, handleChange, isChecked }) => {
+const TodoList = ({ todoList, onRemoveTodo, handleChange, areChecked }) => {
   return (
     <ul>
-      {todoList.map((todo) => {
+      {todoList.map((todo, index) => {
         const { id } = todo;
 
         return (
           <TodoListItem
             key={id}
             todo={todo}
             onRemoveTodo={onRemoveTodo}
-            isChecked={todo.true}
-            handleChange={() => handleChange(id)}
+            idChecked={areChecked[index] ?? undefined}
```

Comment:
all values already start out as `undefined` if we haven't modified them yet, so this fallback is redundant and describes what is already the case
```suggestion
            idChecked={areChecked[index]}
```